### PR TITLE
Add config timeout option

### PR DESCRIPTION
**Pull request by achamayou**
Second part of #5313.

### CODE REVIEW

File: tests/e2e_operations.py [hunk 1]
```diff
@@ -354,6 +355,41 @@ def run_tls_san_checks(args):
...
+    os.remove(os.path.join(start_node_path, "service_cert.pem"))
+    # Save configuration
+    shutil.copy(
+        os.path.join(start_node_path, "0.config.json"),
+        os.path.join(start_node_path, "0.config.json.bak"),
+    )
+    # Replace it with a prefix
+    with open(os.path.join(start_node_path, "0.config.json"), "w") as f:
+        f.write("{")
+    LOG.info("Attempt to start node with a partial config under {start_node_path}")
```
Comment (eddyashton) [inline tests/e2e_operations.py:375]:
```suggestion
    LOG.info(f"Attempt to start node with a partial config under {start_node_path}")
```

Can we test that the node also spins (happily) if there's _no_ file for a few seconds??

File: src/host/main.cpp [hunk 2]
```diff
@@ -102,16 +106,34 @@ int main(int argc, char** argv)
...
+  std::string config_parsing_error = "";
+  do
   {
-    config_json = nlohmann::json::parse(config_str);
-  }
-  catch (const std::exception& e)
+    std::string config_str = files::slurp_string(
+      config_file_path,
+      true /* return an empty string if the file does not exist */);
+    LOG_INFO_FMT("READ CONFIG {}", config_str);
```
Comment (eddyashton) [inline src/host/main.cpp:121]:
```suggestion
```

Looks like debug logging?

File: src/host/main.cpp [hunk 3]
```diff
@@ -102,16 +106,34 @@ int main(int argc, char** argv)
...
+    try
+    {
+      config_json = nlohmann::json::parse(config_str);
+      config_parsing_error = "";
```
Comment (eddyashton) [inline src/host/main.cpp:125]: Should there be a `break` in this path? Why are we still looping if parse succeeded?

File: src/host/main.cpp [hunk 4]
```diff
@@ -102,16 +106,34 @@ int main(int argc, char** argv)
...
+    }
+    catch (const std::exception& e)
+    {
+      config_parsing_error = fmt::format(
+        "Error parsing configuration file {}: {}", config_file_path, e.what());
```
Comment (eddyashton) [inline src/host/main.cpp:130]: Can we add a short sleep in this failed case?

File: tests/e2e_operations.py (hunk 1 above)
Comment (achamayou) [reply tests/e2e_operations.py:375]: Sure

File: src/host/main.cpp (hunk 2 above)
Comment (achamayou) [reply src/host/main.cpp:121]: Thank you, yes

File: src/host/main.cpp (hunk 3 above)
Comment (achamayou) [reply src/host/main.cpp:125]: Yes indeed.

File: src/host/main.cpp (hunk 4 above)
Comment (achamayou) [reply src/host/main.cpp:130]: Sure

File: tests/e2e_operations.py [hunk 5]
```diff
@@ -354,6 +355,45 @@ def run_tls_san_checks(args):
...
+    os.remove(os.path.join(start_node_path, "node.pid"))
+    os.remove(os.path.join(start_node_path, "service_cert.pem"))
+    # Move configuration
+    shutil.move(
+        os.path.join(start_node_path, "0.config.json"),
+        os.path.join(start_node_path, "0.config.json.bak"),
+    )
+    LOG.info("No config at all")
+    assert not os.exists(os.path.join(start_node_path, "0.config.json"))
+    LOG.info("Attempt to start node without a config under {start_node_path}")
```
Comment (eddyashton) [inline tests/e2e_operations.py:374]:
```suggestion
    LOG.info(f"Attempt to start node without a config under {start_node_path}")
```

**Review by eddyashton — APPROVED** [review]

File: tests/e2e_operations.py [hunk 6]
```diff
@@ -354,6 +355,45 @@ def run_tls_san_checks(args):
...
+    shutil.rmtree(os.path.join(start_node_path, "0.ledger"))
+    os.remove(os.path.join(start_node_path, "node.pid"))
+    os.remove(os.path.join(start_node_path, "service_cert.pem"))
+    # Move configuration
+    shutil.move(
+        os.path.join(start_node_path, "0.config.json"),
+        os.path.join(start_node_path, "0.config.json.bak"),
+    )
+    LOG.info("No config at all")
+    assert not os.exists(os.path.join(start_node_path, "0.config.json"))
```
Comment (achamayou) [inline tests/e2e_operations.py:373]:
```suggestion
    assert not os.path.exists(os.path.join(start_node_path, "0.config.json"))
```

**Pull request merged**